lib/decidim/git_backport_manager.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "English"
+
+module Decidim
+  class GitBackportManager
+    def initialize(pull_request_id:, release_branch:, backport_branch:, working_dir: Dir.pwd)
+      @pull_request_id = pull_request_id
+      @release_branch = release_branch
+      @backport_branch = backport_branch
+      @working_dir = working_dir
+    end
+
+    def call
+      Dir.chdir(working_dir) do
+        exit_if_unstaged_changes
+        self.class.checkout_develop
+        sha_commit = sha_commit_to_backport
```

Comment:
I made a small mistake and I did not pull after merging the original PR, so the commit would not locally exist in this case.

I would add the check here as follows:
```suggestion
        sha_commit = sha_commit_to_backport
        unless sha_commit
          exit_with_errors("Could not find commit for pull request #{pull_request_id}. Please make sure you have pulled the latest changes.")
        end
```